sdk/keyvault/tests.yml

```diff
@@ -6,6 +6,8 @@ extends:
     ServiceDirectory: keyvault
     MaxParallel: 5
     TimeoutInMinutes: 180
+    # Keyvault test resources are expensive, and managed HSM modules don't necessarily work in Canary
```

Comment:
Nit:

```suggestion
    # Managed HSM test resources are expensive, and provisioning has not been reliable.
    # Given test coverage on AzureCloud we probably don't need to test in Canary.
```

Reply:
Is it just the HSM resources that are expensive? I thought vaults in general were adding to our cost, hence why I mentioned both.

Reply:
This change has the effect of removing all keyvault tests from canary, not just HSM.

Reply:
Key Vault is per-call pricing and can be, but we could leave that enabled if you find it valuable. Managed HSM is by far the most expensive. I didn't realize this would exclude the whole matrix. We'd only want to exclude Canary for Managed HSM at least.

Reply:
Ok, updated to only skip HSM for canary.